tea.py

```diff
@@ -0,0 +1,674 @@
+import pybamm
+import matplotlib.pyplot as plt
+import numpy as np
+
+
+class Tea(pybamm.BaseModel):
```

Comment:
I don't think this needs to inherit from `BaseModel`. It can just be a new class. I'd also rename it so it's all uppercase, i.e. `TEA()`